coalib/settings/ConfigurationGathering.py

```diff
@@ -109,11 +109,18 @@ def warn_nonexistent_targets(targets, sections, log_printer):
     :param sections:    The sections to search. (Dict.)
     :param log_printer: The log printer to warn to.
     """
+
+    sections_substrings = {
+        substring
+        for sections_name in sections
+        for substring in sections_name.split('.')
+    }
     for target in targets:
-        if target not in sections:
+        if target not in sections_substrings | set(sections):
```

Comment:
also that code was just pseudo code, the most important thing to me is to keep the logic in that place and not generating new artificial targets as you do below IIUC